packages/headless/src/features/product-recommendations/product-recommendations-actions.ts

```diff
@@ -94,36 +108,51 @@ export const getProductRecommendations = createAsyncThunk<
       return rejectWithValue(fetched.error);
     }
     return {
-      recommendations: fetched.success.results.map(mapResultToProductResult),
+      recommendations: fetched.success.results.map((result) =>
+        mapResultToProductResult(result, {
+          additionalFields: state.productRecommendations.additionalFields || [],
+        })
+      ),
       analyticsAction: logProductRecommendations(),
       searchUid: fetched.success.searchUid,
       duration,
     };
   }
 );
 
-const mapResultToProductResult = (result: Result): ProductRecommendation => {
-  const price = result.raw.ec_price as number | undefined;
-  const promoPrice = result.raw.ec_promo_price as number | undefined;
-  const inStock = result.raw.ec_in_stock as string | undefined;
+const mapResultToProductResult = (
+  result: Result,
+  {additionalFields}: {additionalFields: string[]}
+): ProductRecommendation => {
+  const ec_price = result.raw.ec_price as number | undefined;
+  const ec_promo_price = result.raw.ec_promo_price as number | undefined;
+  const ec_in_stock = result.raw.ec_in_stock as string | undefined;
 
   return {
-    sku: result.raw.permanentid!,
-    name: result.raw.ec_name as string,
-    link: result.clickUri,
-    brand: result.raw.ec_brand as string,
-    category: result.raw.ec_category as string,
-    price,
-    shortDescription: result.raw.ec_shortdesc as string,
-    thumbnailUrl: result.raw.ec_thumbnail as string,
-    imageUrls: result.raw.ec_images as string[],
-    promoPrice:
-      promoPrice === undefined || (price !== undefined && promoPrice >= price)
+    permanentid: result.raw.permanentid!,
+    clickUri: result.clickUri,
+    ec_name: result.raw.ec_name as string,
+    ec_brand: result.raw.ec_brand as string,
+    ec_category: result.raw.ec_category as string,
+    ec_price,
+    ec_shortdesc: result.raw.ec_shortdesc as string,
+    ec_thumbnails: result.raw.ec_thumbnails as string,
+    ec_images: result.raw.ec_images as string[],
+    ec_promo_price:
+      ec_promo_price === undefined ||
+      (ec_price !== undefined && ec_promo_price >= ec_price)
+        ? undefined
+        : ec_promo_price,
+    ec_in_stock:
+      ec_in_stock === undefined
         ? undefined
-        : promoPrice,
-    inStock:
-      inStock === undefined ? undefined : inStock.toLowerCase() === 'yes',
-    rating: result.raw.ec_rating as number,
+        : ec_in_stock.toLowerCase() === 'yes' ||
+          ec_in_stock.toLowerCase() === 'true',
```

Comment:
Snuck this in :see_no_evil: 

We saw some real data that put `true` here, so it does little harm to add it

Reply:
Oops ! 

Is that something that is on the team radar as a potential improvement for the catalog service ? 
To "coerce" truthy value ?
I'm not even sure if that's applicable ?

Reply:
I'd say yes, for two reasons:

1- The Catalog source have pre-processing for that, so we can have "standard" values. But people are still using the Push API because we don't support single item updates (yet), so I feel like `yes/true` is _fine_ for now

2- I would expect our eventual back-end API to directly return a boolean and accept both `yes/true` as well

Reply:
This is fine for now but it would be good to solve at indexing so the benefits are available to people using the api directly.  Atm the benefits will only be available to people using the headless controller.